collector/rdma_linux.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !nordma
+// +build !nordma
+
+// The hard work of collecting data from the kernel via the MLNX_OFED interfaces is done by
+// https://github.com/Mellanox/rdmamap
+// by Mellanox. Used under the Apache 2.0 license.
+
+package collector
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Mellanox/rdmamap"
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	rdmaDeviceInclude   = kingpin.Flag("collector.rdma.device-include", "Regexp of rdma devices to include (mutually exclusive to device-exclude).").String()
+	rdmaDeviceExclude   = kingpin.Flag("collector.rdma.device-exclude", "Regexp of rdma devices to exclude (mutually exclusive to device-include).").String()
+	rdmaIncludedMetrics = kingpin.Flag("collector.rdma.metrics-include", "Regexp of rdma stats to include.").Default(".*").String()
+
+	rdmaHwCounters = map[string]string{
+		"roce_slow_restart_cnps":         "RDMA RoCE slow restart CNPS",
```

Comment:
All these need to follow prometheus naming best practices, e.g _total suffix for counters, spelling our abbreviations, including units etc etc. Also where reasonable they should be combined into one metric with labels. Use existing metric naming here as well as https://prometheus.io/docs/practices/naming/ for reference.

Reply:
Please see the latest commit. I’ve updated all metrics and their help to align with the naming convention. I’ve also updated the `metrics.log` in the PR description.